firecrest/BasicClient.py

```diff
@@ -611,14 +618,81 @@ def compress(self, machine: str, source_path: str, target_path: str) -> str:
             additional_headers={"X-Machine-Name": machine},
             data={"targetPath": target_path, "sourcePath": source_path},
         )
-        self._json_response([resp], 201)
+        timeout_str = "Command has finished with timeout signal"
+        if (
+            resp.status_code == 201 or
+            fail_on_timeout or
```

Comment:
Yes, it will be `True` regardless the status code. I was not sure of what is the best order. The idea is:
- if the first request was successful then don't care for `fail_on_timeout`. We just return the response normally
- if the first request failed, but the user has `fail_on_timeout` we let `_json_response` take care of raising the appropriate error
- if the request error failed (but specifically with error code 400 and with the timeout error then we submit the job for compression. The reason that don't directly check the json is that in certain errors (like 500) we may not have a json to check.